SwrveSDK/src/main/java/com/swrve/sdk/messaging/view/SwrveMessageView.java

```diff
@@ -1,251 +1,312 @@
+                    bitmapCache.add(new WeakReference<Bitmap>(imageBitmap));
+                    // Position
+                    RelativeLayout.LayoutParams lparams = new RelativeLayout.LayoutParams(backgroundImage.getWidth(), backgroundImage.getHeight());
+                    lparams.leftMargin = image.getPosition().x;
+                    lparams.topMargin = image.getPosition().y;
+                    lparams.width = backgroundImage.getWidth();
+                    lparams.height = backgroundImage.getHeight();
+                    imageView.setLayoutParams(lparams);
+                    imageView.setImageBitmap(imageBitmap);
+                    imageView.setScaleType(ScaleType.FIT_XY);
+                    // Add to parent
+                    addView(imageView);
+                } else {
+                    loadErrorReasons.add("Could not decode bitmap from file:" + filePath);
+                    break;
+                }
+            }
 
-    /**
-     * Set the install button listener to process message element events.
-     *
-     * @param installButtonListener
-     */
-    public void setInstallButtonListener(ISwrveInstallButtonListener installButtonListener) {
-        this.innerMessageView.setInstallButtonListener(installButtonListener);
-    }
+            for (final SwrveButton button : format.getButtons()) {
+                String filePath = message.getCacheDir().getAbsolutePath() + "/" + button.getImage();
+                if(!SwrveHelper.hasFileAccess(filePath)) {
+                    SwrveLogger.e(LOG_TAG, "Do not have read access to message asset for:" + filePath);
+                    loadErrorReasons.add("Do not have read access to message asset for:" + filePath);
+                    continue;
+                }
+
+                // Load button image
+                final BitmapResult backgroundImage = decodeSampledBitmapFromFile(filePath, screenWidth, screenHeight, minSampleSize);
+                if (backgroundImage != null && backgroundImage.getBitmap() != null) {
+                    Bitmap imageBitmap = backgroundImage.getBitmap();
+                    SwrveButtonView buttonView = createSwrveButton(context, button.getActionType());
+                    bitmapCache.add(new WeakReference<Bitmap>(imageBitmap));
+                    // Position
+                    RelativeLayout.LayoutParams lparams = new RelativeLayout.LayoutParams(backgroundImage.getWidth(), backgroundImage.getHeight());
+                    lparams.leftMargin = button.getPosition().x;
+                    lparams.topMargin = button.getPosition().y;
+                    lparams.width = backgroundImage.getWidth();
+                    lparams.height = backgroundImage.getHeight();
+                    buttonView.setLayoutParams(lparams);
+                    buttonView.setImageBitmap(imageBitmap);
+                    buttonView.setScaleType(ScaleType.FIT_XY);
+                    buttonView.setOnClickListener(new OnClickListener() {
```

Comment:
This logic does not belong in the View component. Can it be moved to the Activity?